src/app/common/hooks/analytics/transactions-analytics.hooks.ts

```diff
@@ -111,3 +113,25 @@ export function useTrackChangedTransactions(
   );
   return result;
 }
+
+function isFirstDeposit(stxBalance: AccountStxBalanceBigNumber): boolean {
+  const storageKey = 'no-deposit';
+  if (stxBalance.total_received.isEqualTo(0)) {
+    localStorage.setItem(storageKey, 'true');
+  } else {
+    if (localStorage.getItem(storageKey) === 'true') {
+      localStorage.removeItem(storageKey);
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
Can we infer a first deposit based on the account history, rather than using localstorage? 

I really think we should avoid direct localstorage operations like this, I've seen localstorage cause bugs so many times.

Reply:
We need to have a state transition: from 0 STX to having some STX to be able to trigger the event.
How would you do it with the account transactions alone ?

Reply:
It's tricky. Would it be accurate enough if we were to just fire an event on every account that has a single STX deposit? 

We'd potentially fire duplicate events per accounts, but isn't it tied to an ID we could filter, so it wouldn't result in inaccurate data?

I'm not sure. If we need this solution, it's okay, but it'll be a much tider implementation when the persistence behaviour is abstracted behind redux etc.

Reply:
It will fire too many events forever, for pretty much all accounts which will count towards our event quota. I think we can quickly ship this simple solution for now. It works, it's just analytics and does not affect other parts. It will also be easier for the user of segment to track.

Reply:
It seems like if my onboarding steps are needing to be refactored to use redux-persist before merging then this should too? Aren't we trying to avoid introducing new uses of localstorage which later need to be refactored?

Reply:
Sure then ! I will change it to redux-persist

Reply:
Does it mean this will ship after we release #2172 ?

Reply:
Yes, we feel that refactoring to redux-persist is better long term.
If you **really really want** to ship it before, I can ship the branch with the localStorage implementation.
Let me know.

Reply:
Oh, no, I can wait. I was just asking to understand the order of the release.